wormhole-connect/src/views/Redeem/Redeem.tsx

```diff
@@ -17,131 +17,157 @@ import Spacer from '../../components/Spacer';
+      return;
+    }
+    let cancelled = false;
+    (async () => {
+      let i = 0;
+      let msgInfo: MessageInfo | undefined;
+      while (msgInfo === undefined && !cancelled) {
+        try {
+          msgInfo = await new Operator().getMessageInfo(
+            route,
+            txData.sendTx,
+            txData.fromChain,
+          );
+        } catch {}
+        if (cancelled) {
+          return;
+        }
+        if (msgInfo !== undefined) {
+          setMessageInfo(msgInfo);
+        } else {
+          await sleep(i < 10 ? 3000 : 30000);
+        }
         i++;
-      } else {
-        clearInterval(initializePoll);
       }
-    }, 1000);
-
-    const poll = setInterval(async () => {
-      if (!this.props.transferComplete) {
-        this.update();
-      } else {
-        clearInterval(poll);
+    })();
+    return () => {
+      cancelled = true;
+    };
+  }, [
+    txData.sendTx,
+    txData.fromChain,
+    messageInfo,
+    setMessageInfo,
+    route,
+    transferComplete,
+  ]);
+
+  // check if VAA has been redeemed
+  useEffect(() => {
+    if (!messageInfo || !txData.toChain || transferComplete) {
+      return;
+    }
+    let cancelled = false;
+    (async () => {
+      let i = 0;
+      let isComplete = false;
+      while (!isComplete && !cancelled) {
+        try {
+          isComplete = await new Operator().isTransferCompleted(
+            route,
+            txData.toChain,
+            messageInfo,
+          );
+        } catch {}
+        if (cancelled) {
+          return;
+        }
+        if (isComplete) {
+          setTransferComplete();
+        } else {
+          await sleep(i < 10 ? 3000 : 30000);
+        }
+        i++;
       }
-    }, 30000);
-  }
-
-  render() {
-    return (
-      <div
-        style={{
-          width: '100%',
-          maxWidth: '700px',
-          display: 'flex',
-          flexDirection: 'column',
-          alignItems: 'center',
-        }}
-      >
-        <PageHeader title="Bridge" back />
-
-        <NetworksTag />
-        <Spacer />
-        <GovernorEnqueuedWarning
-          show={!this.state.messageInfo && this.props.isVaaEnqueued}
-          chain={this.props.txData.fromChain}
-        />
-        <Stepper />
-      </div>
-    );
-  }
+    })();
+    return () => {
+      cancelled = true;
+    };
+  }, [
+    txData.toChain,
+    messageInfo,
+    setMessageInfo,
+    transferComplete,
+    route,
+    setTransferComplete,
+  ]);
+
+  return (
+    <div
+      style={{
+        width: '100%',
+        maxWidth: '700px',
+        display: 'flex',
+        flexDirection: 'column',
+        alignItems: 'center',
+      }}
+    >
+      <PageHeader title="Bridge" back />
+
+      <NetworksTag />
+      <Spacer />
+      <GovernorEnqueuedWarning
+        show={!messageInfo && isVaaEnqueued}
+        chain={txData.fromChain}
+      />
+      <Stepper />
+    </div>
+  );
 }
 
 function mapStateToProps(state: RootState) {
```

Comment:
Is this stuff at the bottom of the file still needed? The redux state and dispatch can now be used directly inside the Redeem component

Reply:
We were going for the minimally invasive fix for this, but agree we should switch to the hooks (and functional components) across the board.